packages/http/test/merge-patch.test.ts

```diff
@@ -0,0 +1,996 @@
+import { Diagnostic, Model, ModelProperty, Program, Type, TypeKind } from "@typespec/compiler";
+import {
+  BasicTestRunner,
+  expectDiagnosticEmpty,
+  expectDiagnostics,
+} from "@typespec/compiler/testing";
+import { $ } from "@typespec/compiler/typekit";
+import { deepStrictEqual, ok } from "assert";
+import { beforeEach, describe, expect, it } from "vitest";
+import {
+  getMergePatchProperties,
+  getMergePatchSource,
+  isMergePatch,
+} from "../src/experimental/merge-patch/helpers.js";
+import { getAllHttpServices } from "../src/operations.js";
+import { HttpOperation, RouteResolutionOptions } from "../src/types.js";
+import {
+  createHttpTestRunner,
+  diagnoseOperations,
+  getOperationsWithServiceNamespace,
+} from "./test-host.js";
+
+let runner: BasicTestRunner;
+beforeEach(async () => {
+  runner = await createHttpTestRunner();
+});
+
+function checkNullableUnion(program: Program, union: Type): boolean {
+  return (
+    $(program).union.is(union) &&
+    union.variants.size === 2 &&
+    [...union.variants.values()].some(
+      (v) => $(program).intrinsic.is(v.type) && v.type.name === $(program).intrinsic.null.name,
+    )
+  );
+}
+
+function getNonNullableType(union: Type): Type | undefined {
+  const program: Program = runner.program;
+  if (!$(program).union.is(union)) return undefined;
+  const values = [...union.variants.values()]
+    .filter((v) => v.type !== $(program).intrinsic.null)
+    .flatMap((variant) => variant.type);
+  if (values.length !== 1) return undefined;
+  return values[0];
+}
+
+function checkProperty(
+  model: Model,
+  name: string,
+  optional: boolean,
+  kind: TypeKind,
+  typeName?: string,
+): ModelProperty {
+  const prop = model.properties.get(name);
+  ok(prop);
+  expect(prop.optional).toBe(optional);
+  deepStrictEqual(prop.type.kind, kind);
+  if (typeName && (prop.type.kind === "Model" || prop.type.kind === "Scalar")) {
+    expect(prop.type.name).toStrictEqual(typeName);
+  }
+  return prop;
+}
+
+function isNullableUnion(property: ModelProperty) {
+  const program: Program = runner.program;
+  deepStrictEqual(property.type.kind, "Union");
+  expect(checkNullableUnion(program, property.type)).toBe(true);
+  return property;
+}
+async function compileAndDiagnoseWithRunner(
+  runner: BasicTestRunner,
+  code: string,
+  options?: RouteResolutionOptions,
+): Promise<[HttpOperation[], readonly Diagnostic[]]> {
+  await runner.compileAndDiagnose(
+    `@service(#{title: "Test Service"}) namespace TestService;
+      ${code}`,
+    {
+      noEmit: true,
+    },
+  );
+  const [services] = getAllHttpServices(runner.program, options);
+  return [services[0].operations, runner.program.diagnostics];
+}
+
+describe("merge-patch: metadata tests", () => {
```

Comment:
would remove the prefix `merge-patch` as its already scopped to the file in vitest